tests/Unit/UtilizationMetricsApisTest.php

```diff
@@ -0,0 +1,425 @@
+      in the Elastic License 2.0.
+
+    For more information or inquiries please visit our website at
+    https://www.canyongbs.com or contact us via email at legal@canyongbs.com.
+
+</COPYRIGHT>
+*/
+
+use App\Models\User;
+
+use function Pest\Laravel\get;
+
+use AdvisingApp\Ai\Models\Prompt;
+use AdvisingApp\Form\Models\Form;
+use AdvisingApp\Task\Models\Task;
+use App\Settings\LicenseSettings;
+use AdvisingApp\Ai\Models\AiThread;
+use AdvisingApp\Alert\Models\Alert;
+use AdvisingApp\Ai\Models\PromptUse;
+use AdvisingApp\Survey\Models\Survey;
+use AdvisingApp\Segment\Models\Segment;
+use App\Http\Middleware\CheckOlympusKey;
+use AdvisingApp\Campaign\Models\Campaign;
+use AdvisingApp\Prospect\Models\Prospect;
+use AdvisingApp\Form\Models\FormSubmission;
+use AdvisingApp\MeetingCenter\Models\Event;
+
+use function Pest\Laravel\withoutMiddleware;
+
+use AdvisingApp\Report\Enums\TrackedEventType;
+use AdvisingApp\Campaign\Models\CampaignAction;
+use AdvisingApp\Survey\Models\SurveySubmission;
+use AdvisingApp\Authorization\Enums\LicenseType;
+use AdvisingApp\Report\Models\TrackedEventCount;
+use AdvisingApp\StudentDataModel\Models\Student;
+use AdvisingApp\ResourceHub\Models\ResourceHubArticle;
+
+it('checks the API returns users', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    User::factory()->count($randomRecords)->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['users'])->toBe($randomRecords);
+});
+
+it('checks the API returns AI Users', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    $licenseSettings = app(LicenseSettings::class);
+
+    $licenseSettings->data->limits->conversationalAiSeats = $randomRecords;
+    $licenseSettings->save();
+
+    User::factory()->count($randomRecords)->licensed([LicenseType::ConversationalAi])->create();
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['ai_users'])->toBe($randomRecords);
+});
+
+it('checks the API returns AI Exchanges', function () {
+    TrackedEventCount::truncate();
+
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = 2;
+
+    TrackedEventCount::factory()->count($randomRecords)->create();
+
+    $totalExchanges = TrackedEventCount::where('type', TrackedEventType::AiExchange)->sum('count');
+
+    $response = get('/api/utilization-metrics');
+
+    $data = $response->json('data');
+
+    $response->assertStatus(200);
+
+    expect($data['ai_exchanges'])->toBe($totalExchanges);
+});
+
+it('checks the API returns Saved AI Chats', function () {
+    withoutMiddleware(CheckOlympusKey::class);
+
+    $randomRecords = random_int(1, 10);
+
+    AiThread::factory()->count($randomRecords)->create();
```

Comment:
The changes I asked to be made in the controller will probably cause this to fail. As you will want to use the `saved` state here so that the `saved_at` is also filled.

Reply:
Done